### PR TITLE
Video Upload Date fix

### PR DESCRIPTION
**Pull request by Bhikule19**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -842,37 +842,87 @@ function display_rich_snippet( $content ) {
 		$args_video = get_option( 'bsf_video' );
 		$video      = '';
 
+		// Generate the schema structure.
```
Comment (sushmak02) [inline functions.php:845]: add this check here - 

if ( ! isset( $post->ID ) ) {
		return;
	}

File: functions.php [hunk 2]
```diff
@@ -842,37 +842,87 @@ function display_rich_snippet( $content ) {
...
+			// Check if $video_date is set and valid.
+			if ( ! empty( $video_date ) ) {
+				// Ensure $timezone is set.
+				if ( isset( $timezone ) ) {
+					try {
+						// Create a DateTime object from the $video_date string.
+						$datetime   = new DateTime( $video_date, new DateTimeZone( $timezone ) ); // Set the timezone to the server's timezone.
+						$uploadDate = $datetime->format( 'Y-m-d\TH:i:sP' ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+					} catch ( Exception $e ) {
+						echo 'Error creating DateTime object: ' . esc_html( $e->getMessage() );
```
Comment (sushmak02) [inline functions.php:868]: add translation support __(

**Review by sushmak02 — COMMENTED** [review]
Check comments on PR

File: functions.php (hunk 1 above)
Comment (akshayurankar48) [reply functions.php:845]: done @sushmak02

File: functions.php [hunk 3]
```diff
@@ -842,37 +842,87 @@ function display_rich_snippet( $content ) {
...
+
+		// Set default timezone and handle video date.
+		if ( '' != trim( $video_date ) ) {
+			// Get the server timezone.
+			$timezone = date_default_timezone_get();
+
+			// Check if $video_date is set and valid.
+			if ( ! empty( $video_date ) ) {
+				// Ensure $timezone is set.
+				if ( isset( $timezone ) ) {
```
[the remark this answers is not in the document]
Comment (akshayurankar48) [reply functions.php:862]: done

**Review by sushmak02 — APPROVED** [review]

File: functions.php [hunk 4]
```diff
@@ -839,40 +839,93 @@ function display_rich_snippet( $content ) {
...
+
+			// Check if $video_date is set and valid, and $timezone is set.
+			if ( ! empty( $video_date ) && isset( $timezone ) ) {
+				try {
+					// Create a DateTime object from the $video_date string.
+					$datetime   = new DateTime( $video_date, new DateTimeZone( $timezone ) ); // Set the timezone to the server's timezone.
+					$uploadDate = $datetime->format( 'Y-m-d\TH:i:sP' ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+				} catch ( Exception $e ) {
+					// Translators: %s is the error message from the exception.
+					echo esc_html( sprintf( __( 'Error creating DateTime object: %s', 'text-domain' ), esc_html( $e->getMessage() ) ) );
```
Comment (sushmak02) [inline functions.php:872]: 'text-domain' should be actual domain of our plugin,

never copy paste anything blindly

Comment (akshayurankar48) [reply functions.php:872]: sorry was checking something in hurry , will be mindful moving forward

**Review by sushmak02 — APPROVED** [review]

**Pull request merged**